File: LeetCode/905/main.py
class Solution:
    def sortArrayByParity(self, arr: List[int]) -> List[int]:
        # arr.sort(key=lambda x: x & 1)
        left, right = 0, len(arr) - 1

        while(left <= right) :
            
            if(arr[left] % 2 > arr[right] % 2) :
                arr[left], arr[right] = arr[right], arr[left]
            
            if(arr[left] % 2 == 0) :
                left += 1
            
            if(arr[right] % 2 == 1) :
                right -= 1
        
        return arr

File: LeetCode/905/main.cpp
class Solution {
public:
  vector<int> sortArrayByParity(vector<int> &arr) {
    // sort(arr.begin(), arr.end(), [](int a, int b) -> bool {
    //     return (a & 1) < (b & 1);
    // });

    int left = 0, right = arr.size() - 1;

    while (left <= right) {
      if (arr[left] % 2 > arr[right] % 2) {
        int temp = arr[left];
        arr[left] = arr[right];
        arr[right] = temp;
      }
      if (arr[left] % 2 == 0) {
        left++;
      }
      if (arr[right] % 2 == 1) {
        right--;
      }
    }

    return arr;
  }
};